product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import odoo.tests.common as common
+
+
+class TestProductSupplierinfoForCustomerSale(common.TransactionCase):
+
+    def setUp(self):
+        super(TestProductSupplierinfoForCustomerSale, self).setUp()
+        self.supplierinfo_model = self.env['product.supplierinfo']
+        self.pricelist_item_model = self.env['product.pricelist.item']
+        self.pricelist_model = self.env['product.pricelist']
+        self.customer = self._create_customer('customer1')
+        self.product = self.env.ref('product.product_product_4')
+        self.supplierinfo = self._create_supplierinfo(
+            'customer', self.customer, self.product)
+        self.pricelist = self._create_pricelist(
+            'Test Pricelist', self.product)
+        self.company = self.env.ref('base.main_company')
+
+    def _create_customer(self, name):
+        """Create a Partner."""
+        return self.env['res.partner'].create({
+            'name': name,
+            'email': 'example@yourcompany.com',
+            'customer': True,
+            'phone': 123456,
+        })
+
+    def _create_supplierinfo(self, type, partner, product):
+        return self.env['product.supplierinfo'].create({
```

Comment:
@xavierjimenez IMHO use `supplierinfo_model`

Reply:
You mean, `self.supplierinfo_model`
+1